activesupport/test/time_zone_test.rb

```diff
@@ -725,6 +725,15 @@ def test_all_uninfluenced_by_time_zone_lookups_delegated_to_tzinfo
     assert_not_includes all_zones, galapagos
   end
 
+  def test_all_not_raises_exception_with_mizzing_tzinfo_data
+    with_missing_tzdata_for("Puerto Rico") do |timezone|
+      assert_nil timezone["Puerto Rico"]
+      assert_nothing_raised do
+        timezone.all
+      end
+    end
+  end
+
```

Comment:
While creating a subclass works as a way of reproducing the error it doesn't really reflect real-life usage so I'd prefer a test that more accurately reflects the error which we can do by changing the mappings for this test using the helper method in the comment below. With that you can then change the test to this:

``` ruby
def test_all_not_raises_exception_with_mizzing_tzinfo_data
  mappings = {
    "Puerto Rico" => "America/Unknown",
    "Pittsburgh"  => "America/New_York"
  }

  with_tz_mappings(mappings) do
    assert_nil ActiveSupport::TimeZone["Puerto Rico"]
    assert_nil ActiveSupport::TimeZone[-9]
    assert_nothing_raised do
      ActiveSupport::TimeZone.all
    end
  end
end
```

Reply:
Thanks, I wasn't particular happy with creating the subclass and now realize that I created the test from the wrong "perspective".